tests/modules/test_bump_versions.py

```diff
@@ -20,14 +26,68 @@ def test_modules_bump_versions_single_module(self):
         with open(env_yml_path, "w") as fh:
             fh.write(new_content)
         version_bumper = nf_core.modules.bump_versions.ModuleVersionBumper(pipeline_dir=self.nfcore_modules)
-        version_bumper.bump_versions(module="bpipe/test")
+        modules = version_bumper.bump_versions(module="bpipe/test")
         assert len(version_bumper.failed) == 0
+        assert [m.component_name for m in modules] == ["bpipe/test"]
 
     def test_modules_bump_versions_all_modules(self):
         """Test updating all modules"""
         version_bumper = nf_core.modules.bump_versions.ModuleVersionBumper(pipeline_dir=self.nfcore_modules)
-        version_bumper.bump_versions(all_modules=True)
+        modules = version_bumper.bump_versions(all_modules=True)
         assert len(version_bumper.failed) == 0
+        assert [m.component_name for m in modules] == ["bpipe/test"]
+
+    @staticmethod
+    def _mock_nf_core_yml(root_dir: Path) -> None:
+        """Mock the .nf_core.yml"""
+        yaml = ruamel.yaml.YAML()
+        yaml.preserve_quotes = True
+        yaml.indent(mapping=2, sequence=2, offset=0)
+        nf_core_yml = NFCoreYamlConfig(nf_core_version=__version__, repository_type="modules", org_path="nf-core")
+        with open(Path(root_dir, ".nf-core.yml"), "w") as fh:
+            yaml.dump(nf_core_yml.model_dump(), fh)
+
+    @staticmethod
+    def _mock_modules(root_dir: Path, modules: List[str]) -> None:
+        """Mock the directory for a given module (or sub-module) for use with `_dryrun`"""
+        nf_core_dir = root_dir / "modules" / "nf-core"
+        for module in modules:
+            if "/" in module:
+                module, sub_module = module.split("/")
+                module_dir = nf_core_dir / module / sub_module
+            else:
+                module_dir = nf_core_dir / module
+            module_dir.mkdir(parents=True)
+            module_main = module_dir / "main.nf"
+            with module_main.open("w"):
+                pass
+
+    def test_modules_bump_versions_multiple_modules(self):
+        """Test updating all modules when multiple modules are present"""
+        # mock the fgbio directory
+        root_dir = Path(tempfile.TemporaryDirectory().name)
+        self._mock_modules(root_dir=root_dir, modules=["fqgrep", "fqtk"])
+        # mock the ".nf-core.yml"
+        self._mock_nf_core_yml(root_dir=root_dir)
+
+        # run it with dryrun to return the modules that it found
+        version_bumper = nf_core.modules.bump_versions.ModuleVersionBumper(pipeline_dir=root_dir)
+        modules = version_bumper.bump_versions(all_modules=True, _dryrun=True)
+        assert sorted([m.component_name for m in modules]) == sorted(["fqgrep", "fqtk"])
```

Comment:
The `test_modules_bump_versions_all_modules` only tests updating all modules when one module is present, which could improperly succeed if only the first module found is updated, hence the addition of this new test